### ISSUE TITLE
Current speed limit is not shown wile navigating

### ISSUE THREAD

**Issue by Talkless** [rinigus/pure-maps#604]
My setup;

Pure Maps 2.9.0
OSM Scout Server 1.17.1
Ubutnu Touch OTA-22
Volla Phone X

Pure Maps settings:
Maps: OSM Scout
Navigation: Every option (including "Snap position to road enabled") is On, Speed limit: Always.

Result: I only see current speed:
![puremaps](https://user-images.githubusercontent.com/292191/163813895-ffd20f56-d32e-4625-a7a8-61a9367d86d5.png)

I was driving this road, which has `maxspeed` set:
https://www.openstreetmap.org/way/60441895#map=12/55.7706/23.0411

**Comment by rinigus** [comment]
That doesn't work on Ubuntu Touch as you don't have DBus communication possible between apps. At least that is my understanding and please correct me if I am wrong (@jonnius). Hence there is no speed limit nor showing current street on UT

**Comment by TadasKrasauskas** [comment]
Same problem using Pure maps in Xperia 10 II (no speed limit sign). 
Sfos 4.4.0.58
Pure maps version 3.0.0-1.5.1

**Comment by rinigus** [comment]
Yes, on SFOS it is due to Sailjail restrictions, similar to UT in this respect. There will be separate version from the next update that disables Sailjail for SFOS.

**Comment by TadasKrasauskas** [comment]
> Yes, on SFOS it is due to Sailjail restrictions, similar to UT in this respect. There will be separate version from the next update that disables Sailjail for SFOS.

Thank you so much. 
Its a best native navigation ;)

**Comment by Talkless** [comment]
@rinigus would local UNIX / IP socket RPC interface help?

Speaking of UT, AppArmor does have ability to write DBus rules, allowing applications to communicate using specifiied DBus services.

Here's some examples for AppArmor built-in abstractions (includes) for allowing quering network information from NetworkManager: https://gitlab.com/apparmor/apparmor/-/blob/master/profiles/apparmor.d/abstractions/dbus-network-manager-strict

Of course, all that needs UT explicit support, to add some sort of DBus permissions for published app...

But maybe just having non-DBus-socket implementation would help? Or that would be considered too much for this "niche" issue?

**Comment by rinigus** [comment]
@Talkless: no, I would prefer to keep it on DBus as it is. As for UT explicit support, there is some already. If it is just a question of packaging then sure, why not. @jonnius works on packaging of Pure Maps for UT, so I would suggest to get in touch with him.

**Comment by Talkless** [comment]
@rinigus I believe I should speak (open issue, etc) with UT developers themselves, how could we allow some apps to do DBus requests, how to specify AppArmor policy, etc. Only when this works in principle we could the talk about proper packaging...

First I'll check if there's AppArmor "DENIED" kernel messages produces by PureMaps. Or this feature is simply disabled on UT at compile time..?

**Comment by rinigus** [comment]
I am a wrong person to ask about UT specifics, as I don't run it on UT. Hopefully @jonnius can reply

**Comment by jonnius** [comment]
I wonder whether DBus should work in our case as OSM Scout Server is unconfined and it is the one offering the service.

**Comment by Talkless** [comment]
I've modified `pure-maps.apparmor` to looke like terminal app:
```
{
    "policy_groups": [],
    "policy_version": 16.04,
    "template": "unconfined"
}
```

But after reboot I still see AppArmor delials:
```
dbus[3308]: apparmor="DENIED" operation="dbus_signal"  bus="session" path="/io/github/rinigus/PureMaps/navigator" interface="io.github.rinigus.PureMaps.navigator" member="destEtaChanged" mask="send" name="org.freedesktop.DBus" pid=6790 label="pure-maps.jonnius_pure-maps_3.1.0" peer_pid=3320 peer_label="unconfined"
```

Probably generated AppArmor profile is cached somewere..?

**Comment by Talkless** [comment]
I've removed `/var/cache/apparmor/click_pure-maps.jonnius_pure-maps_3.1.0` but after reboot still got same denials.

Simpler would probably be to build app myself?

**Comment by jonnius** [comment]
I don't think that changing the apparmor file on the device will take effect. You'd need to rebuild (see [instructions](https://github.com/rinigus/pure-maps/tree/master/packaging/click#readme)).

**Comment by Talkless** [comment]
@rinigus yeah I've managed to build with unconfined AppArmor template.

Had to use `--skip-review` to avoid `(NEEDS REVIEW) 'unconfined' not allowed` error. 

Will try driving some time later.

**Comment by Talkless** [comment]
I have set "Speed limit: Always", but.. I see speed limit extremely rarely, and for a split second only.

For example, I was driving approximately [this route](https://maps.openrouteservice.org/#/directions/197,TG,Lietuva/Architekt%C5%B3%20g.,%C5%A0iauliai,SH,Lietuva/29%20Dubijos%20g,%C5%A0iauliai,SH,Lietuva/99%20Vilniaus%20g,%C5%A0iauliai,SH,Lietuva/data/55,130,32,198,15,97,4,224,38,9,96,59,2,24,5,192,166,6,113,0,184,64,38,124,3,160,13,132,129,88,6,96,19,156,129,217,43,58,128,89,46,96,26,115,202,57,187,246,186,128,70,58,36,5,208,1,192,1,133,128,110,124,148,139,225,40,50,73,73,117,6,247,200,36,184,202,157,187,87,201,50,112,201,149,207,230,111,196,156,133,149,142,241,220,197,233,174,28,185,18,60,50,175,241,213,69,5,5,168,236,136,29,200,4,116,45,221,141,245,60,141,201,121,19,201,165,77,152,200,65,216,64,32,0,29,81,224,33,17,176,113,64,0,188,160,1,109,113,5,196,178,115,160,32,0,205,224,0,109,209,112,65,97,161,224,0,220,144,1,204,1,104,193,145,161,50,65,235,209,27,209,161,209,16,193,218,241,103,32,42,42,231,97,209,96,64,1,125,182,128), but seen speed limits only in [this section](https://maps.openrouteservice.org/#/directions/A12,KS,Lietuva/null/null/196,KS,Lietuva/data/55,130,32,198,15,97,4,224,38,9,96,59,2,24,5,192,166,6,113,0,184,64,38,124,3,160,13,128,118,1,56,40,21,128,22,11,108,32,14,0,25,240,25,157,128,105,174,168,218,72,87,192,17,157,190,38,212,171,81,96,32,55,34,0,174,0,108,149,116,82,190,114,213,234,150,204,42,73,137,106,101,106,212,146,204,164,238,189,248,145,43,76,135,90,135,218,62,162,11,136,8,0,29,83,192,136,141,131,138,0,5,229,0,11,107,140,43,73,237,237,1,0,6,111,4,174,139,130,11,13,15,0,6,228,128,14,96,11,70,12,141,1,226,7,30,128,158,141,14,136,134,10,151,141,89,14,30,19,91,14,139,2,0,11,233,212,0,0) and [this one](https://maps.openrouteservice.org/#/directions/E%C5%BEero%20g.,%C5%A0iauliai,SH,Lietuva/null/null/E%C5%BEero%20g.,%C5%A0iauliai,SH,Lietuva/data/55,130,32,198,15,97,4,224,38,9,96,59,2,24,5,192,166,6,113,0,184,64,38,1,152,3,164,63,124,4,224,1,128,70,1,217,240,21,142,128,217,168,5,142,128,105,28,120,138,137,181,175,142,149,42,140,0,114,16,13,200,128,43,128,27,5,93,229,45,152,185,106,133,210,138,146,111,134,133,26,141,218,22,100,103,159,1,140,90,27,172,122,157,123,32,184,128,128,1,213,60,8,137,176,229,0,5,229,0,11,107,143,74,238,237,1,0,6,111,0,174,139,130,11,13,15,0,6,228,128,14,96,11,70,12,141,2,226,9,30,141,30,141,14,136,134,0,151,134,89,12,28,30,91,14,139,2,0,11,226,212,0).

I did try to drive through second section mentioned today and I can confirm it does show (50) sign for a split second, somewhere around this way: https://www.openstreetmap.org/way/85694140 .

Interestingly, it does not have speed limit set, only way before and after. So maybe it sown next upcoming way's (https://www.openstreetmap.org/way/38826951) limit.

P.S. I have updated speed limits to match reality.

**Comment by rinigus** [comment]
Hard to comment whether it is related to different coverage or not. Will have to test over here with speed limit: always set.

**Comment by Talkless** [comment]
I have successfully reproduced speed notification (for a split second) here:
https://www.openstreetmap.org/way/85694140
and here:
https://www.openstreetmap.org/way/932294416
Some more places like, somewhere around:
https://www.openstreetmap.org/node/457411082
https://www.openstreetmap.org/node/1330470901

So the key point is, there's rather concrete, reproducible, specific "places" where I do get short notifications of speed.

I thought maybe Volla Phone X provided coordinates has rather bad precision and gets mostly discarded? But the fact that I can reproduce these notifications probably disprove that idea? Also, I don't know how all that works so...
